src/api/goods.ts

```
@@ -3,14 +3,43 @@
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+export async function getAll(): Promise<Good[]> {
+  try {
+    const response = await fetch(API_URL);
+
+    if (!response.ok) {
+      throw new Error(`error: ${response.status} ${response.statusText}`);
+    }
+
+    const data: Good[] = await response.json();
+
+    return data;
+  } catch (error) {
+    console.error('error getAll', error);
+    throw error;
+  }
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+export const get5First = async (): Promise<Good[]> => {
+  try {
+    const goods = await getAll();
+
+    return [...goods].sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5);
+  } catch (err) {
+    console.error('error get5First', err);
+
+    return [];
+  }
```

Good job adding a `try...catch` block here for error handling. However, by catching the error from `getAll()` and returning an empty array, you are effectively hiding the error from the `App` component. The `.catch()` block in `handleLoadFirstFive` in `App.tsx` will never be executed because this function's promise will always resolve successfully (with either data or an empty array).

To maintain consistency with `getAll` and allow the UI to handle the error state, consider re-throwing the error here.